app/models/session.rb

```diff
@@ -131,18 +131,21 @@ def create_patient_sessions!
         required_programmes.all? { |programme| patient.vaccinated?(programme) }
       end
 
-    # First we remove patients from any other upcoming sessions.
+    # Mark existing patient sessions for transfer
     unvaccinated_patients.each do |patient|
-      sessions_other_than_self = patient.upcoming_sessions.reject { _1 == self }
-      next if sessions_other_than_self.empty?
+      other_sessions = patient.upcoming_sessions.where.not(id:)
+      next if other_sessions.empty?
 
       patient
         .patient_sessions
-        .where(session: sessions_other_than_self)
-        .find_each(&:destroy_if_safe!)
+        .where(session: other_sessions)
```

Comment:
Do we need to consider patient sessions here that have vaccination records or will that be handled when the transfer is confirmed?

Effectively we can't remove the patient from a session if they've been vaccinated there.

Reply:
I think I'll create `propose_transfer_if_safe!(session)` which handles this

Reply:
In these cases I think we'd still want to move them across to the new session, but we can't remove them from the old session, effectively the patients would exist in both, at least that's how it works currently.

Reply:
Okay, dialled back to the original implementation and we can handle this on the moving UI side.